src/Microsoft.AspNetCore.Mvc.Razor/RazorPageActivator.cs

```diff
@@ -63,7 +63,7 @@ private ViewDataDictionary CreateViewDataDictionary(ViewContext context, PageAct
             if (context.ViewData == null)
             {
                 // Create ViewDataDictionary<TModel>(IModelMetadataProvider, ModelStateDictionary).
-                return (ViewDataDictionary)Activator.CreateInstance(
+                return ViewDataDictionaryFactory.Create(
```

Comment:
I don't want to mix type activation with this. We're also **already** doing a cache lookup in this code, I don't want to add more.